datashare-app/src/main/java/org/icij/datashare/tasks/BatchSearchRunner.java

```diff
@@ -156,11 +144,15 @@ public Integer call() throws Exception {
             logger.info("done batch search {} with success", batchSearch.uuid);
         } catch (ElasticsearchException esEx) {
             logger.error("ES exception while running batch " + taskView.id, esEx);
-            repository.setState(taskView.id, new SearchException(query,
-                    ElasticSearchAdapterException.createFrom(esEx)));
+            SearchException searchException = new SearchException(query,
+                    ElasticSearchAdapterException.createFrom(esEx));
+            repository.setState(taskView.id, searchException);
```

Comment:
I'm wondering if the `setState` is even necessary.
For other tasks, exceptions which are thrown at caught at the `TaskWorkerLoop` level which takes care of handling the error properly. Task status handling shouldn't happen in task themselves, but I might be missing something

Reply:
Ok i get it, the `repository` is a `BatchSearchRepository` and not a `TaskRepository`